# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1400862005 14400
#      Fri May 23 12:20:05 2014 -0400
# Branch stable
# Node ID 4293f49795d930850d16e855255f1e696f56c192
# Parent  5e99fd6ea758825aa9c46138769248a7933c8875
don't crash on empty script files (bug #42064)

* lex.ll (<INPUT_FILE_START><<EOF>>): New pattern.

diff --git a/libinterp/parse-tree/lex.ll b/libinterp/parse-tree/lex.ll
--- a/libinterp/parse-tree/lex.ll
+++ b/libinterp/parse-tree/lex.ll
@@ -275,16 +275,28 @@ ANY_INCLUDING_NL (.|{NL})
     // as the first token.
     curr_lexer->reading_script_file = true;
 
     curr_lexer->pop_start_state ();
 
     return curr_lexer->show_token (INPUT_FILE);
   }
 
+<INPUT_FILE_START><<EOF>> {
+    curr_lexer->lexer_debug ("<INPUT_FILE_START><<EOF>>");
+
+    // May be reset later if we see "function" or "classdef" appears
+    // as the first token.
+    curr_lexer->reading_script_file = true;
+
+    curr_lexer->pop_start_state ();
+
+    return curr_lexer->show_token (INPUT_FILE);
+  }
+
 %{
 // Help and other command-style functions.
 %}
 
 <COMMAND_START>{NL} {
     curr_lexer->lexer_debug ("<COMMAND_START>{NL}");
 
     curr_lexer->input_line_number++;
